### ISSUE TITLE
TimeoutOverflowWarning / Will explode incredibly if interval > 24d

### ISSUE THREAD

**Issue by zkxwpdam** [iccicci/rotating-file-stream#34]
**rotating-file-stream** v1.3.10
**node** v10.14.1
**OS** windows10
# Issue

I tried this settings for at least monthly logs since my app is not expected to produce so much log
```
rfs('foo.log', {
  size: '10M', // rotate every 10 MegaBytes written
  interval: '30d', // rotate 30 days
  compress: 'gzip', // compress rotated files
});
```

and Node spills out millions of

```
(node:1684) TimeoutOverflowWarning: 2155129676 does not fit into a 32-bit signed integer.
Timeout duration was set to 1.
(node:1684) TimeoutOverflowWarning: 2155129670 does not fit into a 32-bit signed integer.
Timeout duration was set to 1.
(node:1684) TimeoutOverflowWarning: 2155129667 does not fit into a 32-bit signed integer.
Timeout duration was set to 1.
(node:1684) TimeoutOverflowWarning: 2155129659 does not fit into a 32-bit signed integer.
Timeout duration was set to 1.
(node:1684) TimeoutOverflowWarning: 2155129636 does not fit into a 32-bit signed integer.
Timeout duration was set to 1.
(node:1684) TimeoutOverflowWarning: 2155129634 does not fit into a 32-bit signed integer.
Timeout duration was set to 1.

```
and a bazillion files are being created (just for illustration)

```
       LastWriteTime Length Name
       ------------- ------ ----
06.01.2019     01:21     20 20190106-0000-01-foo.log
06.01.2019     01:21     20 20190106-0000-01-banana.log
06.01.2019     01:21     20 20190106-0000-01-bar.log
06.01.2019     01:21     20 20190106-0000-01-apple.log
06.01.2019     01:21     20 20190106-0000-02-foo.log
06.01.2019     01:21     20 20190106-0000-02-banana.log
06.01.2019     01:21     20 20190106-0000-02-bar.log
06.01.2019     01:21     20 20190106-0000-02-apple.log
06.01.2019     01:21     20 20190106-0000-03-foo.log
06.01.2019     01:21     20 20190106-0000-03-banana.log
06.01.2019     01:21     20 20190106-0000-03-bar.log
06.01.2019     01:21     20 20190106-0000-03-apple.log
06.01.2019     01:21     20 20190106-0000-04-foo.log
06.01.2019     01:21     20 20190106-0000-04-banana.log
06.01.2019     01:21     20 20190106-0000-04-bar.log
06.01.2019     01:21     20 20190106-0000-04-apple.log
06.01.2019     01:21     20 20190106-0000-05-foo.log
06.01.2019     01:21     20 20190106-0000-05-banana.log
06.01.2019     01:21     20 20190106-0000-05-bar.log
06.01.2019     01:21     20 20190106-0000-05-apple.log
and so on
```
# Cause

After a bit fiddling I figured it out ...
30 days is probably a very big value one sane developer would ever use 😱 
and the node error looked fishy already

```
(node:1684) TimeoutOverflowWarning: 2155129450 does not fit into a 32-bit signed integer.
Timeout duration was set to 1.
```
> 2,147,483,647 is the maximum positive value for a 32-bit signed binary integer in computing

So quick math here for how many **d** you could set option `interal` to
`2,147,483,647 / (60 * 60 * 24) = 24,855` 
Seems a bit much days for crashing with 30 already ...

Throw in a divide by 1000 for good measure 
(I am quite sure it is due to `setInterval()` being in milliseconds 😜) 
and you end up at a maximum of **24.8d** as value

Trying with `interval: '24d'` everything works as expected ✅ 
and `interval: '25d'` explodes 💥 😏 

Maybe check the value and throw error 👍 but better do not let it run wild!

**Comment by iccicci** [comment]
Hi @zkxwpdam ,

I just released version 1.4.0 which should hopefully fix this issue.
I added some unit tests to check it, but to do a real test obviously requires at least 25 days.
I added **M** (month) unit for rotation _interval_ option; I would be glad if you want to use it and give me some feedback in case of other errors.

Thank you for interesting in **rotating-file-stream**,
iCC

**Comment by iccicci** [comment]
Closing due to lack of feedback
